### PR TITLE
feat: add custom metadata encoder for data store

### PR DESCRIPTION
**Pull request by gabrielfs7**
Add custom metadata encoder

- [x] Fix and add tests
- [x] Confirm with developers new JSON format.

https://oat-sa.atlassian.net/browse/ADF-768

### CODE REVIEW

File: manifest.php [hunk 1]
```diff
@@ -79,5 +79,5 @@
     ],
     'e2ePrerequisiteActions' => [
         BuildE2eConfiguration::class
-    ],
+    ]
```
Comment (shpran) [inline manifest.php:82]: image

File: composer.json [hunk 2]
```diff
@@ -22,7 +22,7 @@
     "require": {
         "oat-sa/oatbox-extension-installer": "~1.1||dev-master",
         "oat-sa/generis": ">=14.0.0",
-        "oat-sa/tao-core": ">=48.56.0",
+        "oat-sa/tao-core": ">=48.57.0",
```
Comment (gabrielfs7) [inline composer.json:25]: - [ ] Update to prover version before merge.

File: manifest.php (hunk 1 above)
Comment (gabrielfs7) [reply manifest.php]: look again ;)

**Review by shpran — APPROVED** [review]
- [x] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [ ] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

**Review by hectoras — APPROVED** [review]

**Comment by AntonNeviarouski** [comment]
QA verifed

**Pull request merged**